components/ruby/lib/license_acceptance/product_builder.rb

```diff
@@ -0,0 +1,45 @@
+# We were getting too many args in the Product initialize method so switching to the builder
+# pattern. We also need to now validate the license_name
+require_relative("product")
+require_relative("license_list")
+
+module LicenseAcceptance
+  class ProductBuilder
+    def self.build
+      builder = new
+      yield(builder)
+      builder.product
+    end
```

Comment:
Any reason not to put this method on Product and call use the accessors as the interface?

Reply:
Well, the Builder design pattern has them as separate classes. I'm not sure Builder is really needed here, though - I think just switching to a constructor with keyword args would be simpler and would involve less overhead in calling it.